Sources/iOS/OAuth2WebViewController.swift

```diff
@@ -186,62 +187,73 @@ open class OAuth2WebViewController: UIViewController, UIWebViewDelegate {
 	
 	
 	// MARK: - Web View Delegate
-	
-	open func webView(_ webView: UIWebView, shouldStartLoadWith request: URLRequest, navigationType: UIWebViewNavigationType) -> Bool {
+
+	open func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction, decisionHandler: @escaping (WKNavigationActionPolicy) -> Swift.Void)
+	{
+		let request = navigationAction.request
+
 		guard let onIntercept = onIntercept else {
-			return true
+			decisionHandler(.allow)
+			return
 		}
-		
+
 		// we compare the scheme and host first, then check the path (if there is any). Not sure if a simple string comparison
 		// would work as there may be URL parameters attached
 		if let url = request.url, url.scheme == interceptComponents?.scheme && url.host == interceptComponents?.host {
 			let haveComponents = URLComponents(url: url, resolvingAgainstBaseURL: true)
 			if let hp = haveComponents?.path, let ip = interceptComponents?.path, hp == ip || ("/" == hp + ip) {
-				return !onIntercept(url)
+				if onIntercept(url) {
+					decisionHandler(.cancel)
+				}
+				else {
+					decisionHandler(.allow)
+				}
 			}
 		}
-		
-		return true
+
+		decisionHandler(.allow)
 	}
-	
-	open func webViewDidStartLoad(_ webView: UIWebView) {
-		if "file" != webView.request?.url?.scheme {
+
+	open func webView(_ webView: WKWebView, didStartProvisionalNavigation navigation: WKNavigation!)
+	{
+		if "file" != webView.url?.scheme {
 			showLoadingIndicator()
 		}
 	}
-	
-	/* Special handling for Google's `urn:ietf:wg:oauth:2.0:oob` callback */
-	open func webViewDidFinishLoad(_ webView: UIWebView) {
+
+	open func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!)
+	{
```

Comment:
I'd keep the comment to know why the delegate has been implemented. Plus opening bracket.